solver.rb

```diff
@@ -0,0 +1,12 @@
+class Solver
+    def factorial(number)
+        if number < 0
+            raise ArgumentError
+        end
+        if number == 0
+            1
+        else
+            number * factorial(number - 1)
+        end
+    end
+end
```

Comment:
- [ ] Good job so far :+1: but currently all your tests are failing. Kindly fix that and make sure your tests are passing before submitting for a review :smile: 
- Your tests:
![Screenshot from 2022-07-28 13-16-33](https://user-images.githubusercontent.com/85113567/181502971-568be595-e91c-4ec0-b579-dc8d2425a71e.png)